src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -103,11 +103,18 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
                     }
                 };
 
-                result = await _requestInvoker.RequestServerAsync<CompletionParams, SumType<CompletionItem[], CompletionList>?>(
+                // We want to deserialize to VSCompletionItem type to make sure we don't lose extra information like icons.
+                var response = await _requestInvoker.RequestServerAsync<CompletionParams, SumType<VSCompletionItem[], CompletionList>?>(
                     Methods.TextDocumentCompletionName,
                     serverKind,
                     completionParams,
                     cancellationToken).ConfigureAwait(false);
+
+                if (response.HasValue)
+                {
+                    // Cast VSCompletionItem back to CompletionItem
```

Comment:
So their completion item type doesn't support completion icons?

Reply:
Their `VSCompletionItem` type supports icons but not the `CompletionItem` type, https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=%2Fsrc%2Fproduct%2FLanguageServer.Protocol.Extensions%2FVSCompletionItem.cs

Reply:
ooooo VSCompletionItem is an extension of the normal CompletionItem type/spec. I see I see. Awesome!